File: LOO 2°Parcial/Tarea 2/Biblioteca Personal/Biblioteca Personal/Magazine.cpp
#include "Magazine.h"
Magazine::Magazine() : Caracteristicas() {
	this->volumen = 1;
}
Magazine::Magazine(string title, int ci, int year, int volumen) : Caracteristicas(title, ci, year) {
	this->setVol(volumen);
}

void Magazine::setRevista(string title, int ci, int year, int volumen) {
	this->volumen = volumen;
	this->title = title;
	this->ci = ci;
	this->year = year;
}

void Magazine::setVol(int volumen) {
	this->volumen = volumen;
}
int Magazine::getVol(void){ return this->volumen; }

void Magazine::showDetails(void) {
	Caracteristicas::imprimir();
	cout << "\nNo de volumen: " << this->volumen<< "\n";
}

File: LOO 2°Parcial/Tarea 2/BibliotecaJava/src/bibliotecajava/Magazine.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */

package bibliotecajava;
import bibliotecajava.Caracteristicas;
/**
 * 
 * @author Dimitri Medina Novelo
 */
public class Magazine extends Caracteristicas   {
    protected int volumen;
    
    public Magazine()
        {
            super();
            this.volumen = 0;
        }
    
    public Magazine(String title, int ci, int year, int volumen)
        {
            super( title,  ci,  year);
            this.volumen = volumen;
        }
    public int getVol() {return this.volumen;}
    public void setVol(int value) { this.volumen = volumen < 0 ? 0 : volumen;}
    
    public void setRevista(String title, int ci, int year, int volumen) 
    {
	this.volumen = volumen;
	this.title = title;
	this.ci = ci;
	this.year = year;
    }
    
    public void showDetails()
        {
            super.imprimir(); 
            System.out.print("Volumen: " + this.volumen + "\n");
        }
}
